superagi/llms/ooba_booga.py

```diff
@@ -0,0 +1,106 @@
+from ooba_api import LlamaInstructPrompt, OobaApiClient, Parameters
+
+from superagi.config.config import get_config
+from superagi.llms.base_llm import BaseLlm
+
+DEFAULT_MAX_MODEL_TOKEN_LIMIT = int(get_config("MAX_MODEL_TOKEN_LIMIT") or 2048)
+
+
+class OobaBooga(BaseLlm):
+    """
+    LLM for OoobaBooga's Text Generation Web UI
+
+    You would typically run this locally. Even though it has a web front-end, this will use the API interface
+    """
+
+    def __init__(
+        self,
+        model: str | None = None,
+        url: str = get_config("OOBA_URL"),
+        api_key: str | None = None,
+        temperature: float = 0.05,
+        max_tokens: int = DEFAULT_MAX_MODEL_TOKEN_LIMIT,
+        top_p: float = 0.5,
+        top_k: int = 4,
+        typical_p: int = 1,
+        repetition_penalty: float = 1.0,
+        do_sample: bool = True,
+        instruct: bool = True,
```

Comment:
I'm wondering if it makes more sense to have this similar to Django and just have people create a `Parameters` object rather than doing this shuffling and copying of options that may change over time